src/components/button/button.component.js

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Icon from '../icon';
+import StyledButton, { StyledButtonSubtext } from './button.style';
+import Link from '../link';
+import tagComponent from '../../utils/helpers/tags';
+import OptionsHelper from '../../utils/helpers/options-helper';
+
+const Button = (props) => {
+  const {
+    as,
+    buttonType,
+    disabled,
+    iconPosition,
+    theme,
+    forwardRef,
+    href,
+    ...rest
+  } = props;
+
+  if (props.subtext.length > 0 && props.size !== 'large') {
+    throw new Error('subtext prop has no effect unless the button is large');
+  }
+
+  return (
+    <StyledButton
+      as={ href ? 'div' : 'button' }
+      disabled={ disabled }
+      buttonType={ buttonType || as }
+      role='button'
+      iconPosition={ iconPosition }
+      legacyColorVariant={ theme }
+      { ...tagComponent('button', props) }
+      { ...rest }
+      ref={ forwardRef }
+    >
+      { renderChildren(props) }
+    </StyledButton>
+  );
+};
+
+function renderChildren(props) {
+  const {
+    disabled,
+    iconType,
+    iconPosition,
+    size,
+    subtext,
+    children,
+    href,
+    to
+  } = props;
+  let buttonContent = (
+    <span>
+      <span data-element='main-text'>{ children }</span>
+      { size === 'large' && <StyledButtonSubtext data-element='subtext'>{ subtext }</StyledButtonSubtext> }
+    </span>
+  );
+
+  if (!disabled && (href || to)) {
+    buttonContent = <Link href={ href } to={ to }>{buttonContent}</Link>;
+  }
+
+  return (
+    <>
+      { iconType && iconPosition === 'before' && <Icon type={ iconType } /> }
+      {buttonContent}
```

Comment:
Does this mean the text within the button is the only clickable bit? The entire button should behave as the link.

Reply:
![Screenshot 2019-05-09 at 15 02 10](https://user-images.githubusercontent.com/1668080/57459591-70bccf80-726b-11e9-9b91-06c643bcb790.png)

^ not like this

Reply:
fixed